src/mscorlib/shared/System/StringComparer.cs

```diff
@@ -244,17 +323,22 @@ public override int GetHashCode(string obj)
                 throw new ArgumentNullException(nameof(obj));
 #endif
             }
-            return obj.GetHashCode();
+            return TextInfo.GetHashCodeOrdinalIgnoreCase(obj);
         }
 
         // Equals/GetHashCode methods for the comparer itself. 
-        public override bool Equals(object obj) => obj is OrdinalComparer;
-        public override int GetHashCode() => nameof(OrdinalComparer).GetHashCode();
+        public override bool Equals(object obj) => obj is OrdinalCaseSensitiveComparer;
```

Comment:
Currently `OrdinalIgnoreCaseComparer` and `OrdinalCaseSensitiveComparer` do not inherit from `OrdinalComparer` but from `StringComparer`. In the example you mentioned `EnumEqualityComparer  doesn't inherit from `ObjectEqualityComparer` either.

I don't think they should inherit `OrdinalComparer` as we are loosing optimizations with it and also increasing memory footprint because of the _ignoreCase field.

Do we need the inheritance just because of the Equality check?